src/status_im2/contexts/chat/messages/view.cljs

```diff
@@ -17,9 +17,12 @@
 (defn navigate-back-handler
   []
   (when (and (not @navigation.state/curr-modal) (= (get @re-frame.db/app-db :view-id) :chat))
-    (rn/hw-back-remove-listener navigate-back-handler)
-    (rf/dispatch [:chat/close])
-    (rf/dispatch [:navigate-back])
+    (if (get @re-frame.db/app-db :bottom-sheet/show?)
```

Comment:
we should never access re-frame app-db directly, also photo selector should be a screen